src/edu/stanford/dlss/was/WasapiDownloaderSettings.java

```diff
@@ -3,48 +3,173 @@
+  public static final String JOB_ID_PARAM_NAME = "jobId";
+  public static final String CRAWL_START_AFTER_PARAM_NAME = "crawlStartAfter";
+  public static final String CRAWL_START_BEFORE_PARAM_NAME = "crawlStartBefore";
+
+  private HelpFormatter helpFormatter = null;
+  private Options wdsOpts = null;
   private Properties settings = null;
+  private String helpAndSettingsMessageString = null;
 
+  public WasapiDownloaderSettings(String settingsFileLocation, String[] args) throws LoadSettingsException {
+    try {
+      loadPropertiesFile(settingsFileLocation);
+      setupArgOptions();
+      parseArgsIntoSettings(args);
 
-  public WasapiDownloaderSettings(String settingsFileLocation, String[] args) throws IOException {
-    loadPropertiesFile(settingsFileLocation);
-    //FIXME: parse args for collectionId, jobId, crawlStartAfter, crawlStartBefore.  store in settings var.
+      //TODO: validate settings state?  needed args present?  formatting of e.g. dates?
+    } catch(IOException e) {
+      throw new LoadSettingsException("Error reading properties file: " + e.getMessage(), e);
+    } catch(ParseException e) {
+      throw new LoadSettingsException("Error parsing command line arguments: " + e.getMessage(), e);
+    }
   }
 
 
+  public boolean shouldDisplayHelp() {
+    if(settings.getProperty(HELP_PARAM_NAME) != null) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
   public String baseUrlString() {
-    return settings.getProperty("baseurl");
+    return settings.getProperty(BASE_URL_PARAM_NAME);
   }
 
   public String username() {
-    return settings.getProperty("username");
+    return settings.getProperty(USERNAME_PARAM_NAME);
   }
 
   public String password() {
-    return settings.getProperty("password");
+    return settings.getProperty(PASSWORD_PARAM_NAME);
   }
 
   public String collectionId() {
-    return null; //FIXME: should come from command line args (stored in settings var)
+    return settings.getProperty(COLLECTION_ID_PARAM_NAME);
   }
 
   public String jobId() {
-    return null; //FIXME: should come from command line args (stored in settings var)
+    return settings.getProperty(JOB_ID_PARAM_NAME);
   }
 
   // e.g. 2014-01-01, see https://github.com/WASAPI-Community/data-transfer-apis/tree/master/ait-reference-specification#paths--examples
   public String crawlStartAfter() {
-    return null; //FIXME: should come from command line args (stored in settings var)
+    return settings.getProperty(CRAWL_START_AFTER_PARAM_NAME);
   }
 
   // e.g. 2014-01-01, see https://github.com/WASAPI-Community/data-transfer-apis/tree/master/ait-reference-specification#paths--examples
   public String crawlStartBefore() {
-    return null; //FIXME: should come from command line args (stored in settings var)
+    return settings.getProperty(CRAWL_START_BEFORE_PARAM_NAME);
+  }
+
+  public String getHelpAndSettingsMessageString() {
+    if(helpAndSettingsMessageString == null) {
```

Comment:
`if (`, curly braces

also rename method if rename var above ...

The blank line isn't needed, IMO.   But all this style stuff isn't super important.